server/command.go

```diff
@@ -0,0 +1,233 @@
+
+	show := model.NewAutocompleteData("show", "", "Show MS Teams linked channel")
+	cmd.AddCommand(show)
+
+	return cmd
+}
+
+func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	split := strings.Fields(args.Command)
+	command := split[0]
+	var parameters []string
+	action := ""
+	if len(split) > 1 {
+		action = split[1]
+	}
+	if len(split) > 2 {
+		parameters = split[2:]
+	}
+
+	if command != "/"+msteamsCommand {
+		return &model.CommandResponse{}, nil
+	}
+
+	if action == "link" {
+		return p.executeLinkCommand(c, args, parameters)
+	}
+
+	if action == "unlink" {
+		return p.executeUnlinkCommand(c, args)
+	}
+
+	if action == "show" {
+		return p.executeShowCommand(c, args)
+	}
+
+	return cmdError(args.ChannelId, "Unknown command. Valid options: link, unlink and show.")
+}
+
+func (p *Plugin) executeLinkCommand(c *plugin.Context, args *model.CommandArgs, parameters []string) (*model.CommandResponse, *model.AppError) {
+	if len(parameters) < 2 {
+		return cmdError(args.ChannelId, "Invalid link command, please pass the MS Teams team id and channel id as parameters.")
+	}
+
+	if !p.checkEnabledTeamByTeamId(args.TeamId) {
+		return cmdError(args.ChannelId, "This teams is not enabled for MS Teams sync.")
+	}
+
+	channel, appErr := p.API.GetChannel(args.ChannelId)
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the current channel information.")
+	}
+
+	canLinkChannel := channel.Type == model.ChannelTypeOpen && p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PermissionManagePublicChannelProperties)
+	canLinkChannel = canLinkChannel || (channel.Type == model.ChannelTypePrivate && p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PermissionManagePrivateChannelProperties))
+	if !canLinkChannel {
+		return cmdError(args.ChannelId, "Unable to link the channel, you has to be a channel admin to link it.")
+	}
+
+	channelsLinkedData, appErr := p.API.KVGet("channelsLinked")
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the linked channels information, please try again.")
+	}
+	channelsLinked := map[string]ChannelLink{}
+	json.Unmarshal(channelsLinkedData, &channelsLinked)
+	_, ok := channelsLinked[channel.TeamId+":"+channel.Id]
+	if ok {
+		return cmdError(args.ChannelId, "A link for this channel already exists, please remove unlink the channel before you link a new one")
+	}
+
+	_, err := p.msteamsAppClient.GetChannel(parameters[0], parameters[1])
+	if err != nil {
+		return cmdError(args.ChannelId, "MS Teams channel not found.")
+	}
+
+	channelsLinked[channel.TeamId+":"+channel.Id] = ChannelLink{
+		MattermostTeam:    channel.TeamId,
+		MattermostChannel: channel.Id,
+		MSTeamsTeam:       parameters[0],
+		MSTeamsChannel:    parameters[1],
+	}
+
+	channelsLinkedData, err = json.Marshal(channelsLinked)
+	if err != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
```

Comment:
We should log errors whenever possible

Reply:
I'm not sure how much of this needs server logging because this is interacting directly with the user, so the message is going to be available directly to the user.